presidio-analyzer/analyzer/analyzer_engine.py

```diff
@@ -1,47 +1,54 @@
-import logging
-import os
+import uuid
 
 import analyze_pb2
 import analyze_pb2_grpc
 import common_pb2
 
 from analyzer import RecognizerRegistry
+from analyzer.logger import Logger
+from analyzer.app_tracer import AppTracer
 from analyzer.nlp_engine import SpacyNlpEngine
 
-loglevel = os.environ.get("LOG_LEVEL", "INFO")
-logging.basicConfig(
-    format='%(asctime)s:%(levelname)s:%(message)s', level=loglevel)
 
 DEFAULT_LANGUAGE = "en"
+logger = Logger()
 
 
 class AnalyzerEngine(analyze_pb2_grpc.AnalyzeServiceServicer):
 
     def __init__(self, registry=RecognizerRegistry(),
-                 nlp_engine=SpacyNlpEngine()):
+                 nlp_engine=SpacyNlpEngine(),
+                 ml_tracer=AppTracer()):
         # load nlp module
         self.nlp_engine = nlp_engine
         # prepare registry
         self.registry = registry
         # load all recognizers
         registry.load_predefined_recognizers()
+        self.ml_tracer = ml_tracer
```

Comment:
consider renaming. We don't only trace ml models. We do it for the entire system.